### ISSUE TITLE
Version 1.2.2 crashes with Segmentation fault

### ISSUE THREAD

**Issue by ihopciklumer** [ruby-concurrency/concurrent-ruby#1019]
```
* Operating system:                Mac Sonoma 14.2
* Ruby implementation:             Ruby 2.7.1p83 (2020-03-31 revision a0c7c23c9c)
* `concurrent-ruby` version:       1.2.2
* `concurrent-ruby-ext` installed:  no
* `concurrent-ruby-edge` used:     no
```
Previously used version 1.1.10 but after updating to 1.2.2 get Segmentation fault at 0x0000000027142620

**Comment by eregon** [comment]
We'll need more information than that:
* the full output 
* the Crash Report log file under DiagnosticReports directory

You said
```
`concurrent-ruby-ext` installed:  no
```
in which case it cannot be a bug of concurrent-ruby, and should instead be reported (with as much information as possible to https://bugs.ruby-lang.org/)

**Comment by bensheldon** [comment]
To add a little more details why we're unable to help :-( 

- if you're only using the Ruby version (not the c-ext) of Concurrent Ruby, any segmentation fault (e.g. the Ruby VM crashing) would be the responsibility of the Ruby VM. 
- you're using Ruby 2.7.1, and the 2.7.x release is EOL (March, 2023), and the most recent patch which is Ruby 2.7.8 (so it's possible the segfault has been fixed in later patched versions of Ruby 2.7.x)
